providers/theme.js

```diff
@@ -25,9 +25,26 @@ const ThemeContext = React.createContext({
   toggleTheme: () => {},
 });
 
+const getInitialTheme = () => {
+  if (isServerSide()) {
+    return LIGHT_THEME;
+  }
+
+  const prefersDarkScheme = window.matchMedia('(prefers-color-scheme: dark)').matches;
```

Comment:
Not a huge deal, but I wonder what will happen if you have conflict between your system settings and what is saved in local storage?

Reply:
I totally missed the other `useEffect`, I just combined those so that it only sets the default if there's not a saved theme in local settings.